core/src/ledger_cleanup_service.rs

```diff
@@ -100,43 +100,64 @@ impl LedgerCleanupService {
         root: Slot,
         max_ledger_shreds: u64,
     ) -> (bool, Slot, u64) {
-        let mut total_slots = Vec::new();
-        let mut iterate_time = Measure::start("iterate_time");
-        let mut total_shreds = 0;
-        for (i, (slot, meta)) in blockstore.slot_meta_iterator(0).unwrap().enumerate() {
-            if i == 0 {
-                debug!("purge: searching from slot: {}", slot);
-            }
-            // Unrooted slots are not eligible for cleaning
-            if slot > root {
-                break;
-            }
-            // This method not exact since non-full slots will have holes
-            total_shreds += meta.received;
-            total_slots.push((slot, meta.received));
-        }
-        iterate_time.stop();
+        // TODO: de-dupe this constant
+        let data_shred_cf_name = "data_shred".to_string();
```

Comment:
This needs to be addressed before shipping; a few options:
1) Move this logic to blockstore_db
2) Expose the constant(s) from blockstore_db (currently private)
3) Create an accessor to query this value from `Blockstore`

I think 1) is my least favorite, 2) is quick/easy so kind of leaning that way

Reply:
Option 2 looks good to me.  As a clean-up service, LedgerCleanupService should be allowed to know the status of each column family and have the ability to query CF-specific status.